secure_message/repository/database.py

```diff
@@ -78,12 +78,49 @@ def serialize(self, user, body_summary=False):
                    '_links': '',
                    'labels': []}
 
-        self._populate_to_and_from(user, message)
+        if user.is_internal:
+            self._populate_to_from_and_labels_internal_user(message)
+        else:
+            self._populate_to_from_and_labels_respondent(user, message)
 
         self._populate_events(message)
 
         return message
 
+    def _populate_to_from_and_labels_internal_user(self, message):
+        """populate the labels and to and from if the user is internal"""
+        try:
+            if message['from_internal']:
+                respondent = [x.actor for x in self.statuses if x.label == Labels.INBOX.value or x.label == Labels.DRAFT_INBOX.value][0]
+            else:
+                respondent = [x.actor for x in self.statuses if x.label == Labels.SENT.value or x.label == Labels.DRAFT.value][0]
+        except IndexError:
+            logger("Could not determine respondent from message", msg_id=message["msg_id"])
+            raise
+        for row in self.statuses:
+            if row.actor != respondent:
+                message['labels'].append(row.label)
+            self._add_to_and_from(message, row)
+
+    def _populate_to_from_and_labels_respondent(self, user, message):
+        """Populate labels and to and from if teh user is a respondent"""
+        for row in self.statuses:
+            if row.actor == user.user_uuid:
+                message['labels'].append(row.label)
+            self._add_to_and_from(message, row)
+
+    @staticmethod
+    def _add_to_and_from(message, row):
+        """Populate the message to and from"""
+        if row.label == Labels.INBOX.value:
+            message['msg_to'].append(row.actor)
+        elif row.label == Labels.SENT.value:
```

Comment:
Could simplify this if else statement with the use of in

if row.label in [Labels.INBOX.value, Labels.DRAFT_INBOX]:
            message['msg_to'].append(row.actor)
elif row.label ins [Labels.SENT.value, Labels.DRAFT]:

Reply:
Yep , good call , changing